src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
                 // get inlet and outlet node number from equipment list
                 CurrentModuleObject = "AirLoopHVAC:OutdoorAirSystem:EquipmentList";
                 for (int CompNum = 1; CompNum <= state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).NumComponents; ++CompNum) {
-                    std::string CompType = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentType(CompNum);
-                    std::string CompName = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum);
+                    auto &CompType = state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentType(CompNum);
```

Comment:
To get the full effect of this change, my suggestion is to look at both the develop [version](https://github.com/NREL/EnergyPlus/blob/c8eacd64f9f2ba2d7bd64c2617f4d2474436aefc/src/EnergyPlus/AirLoopHVACDOAS.cc#L399) and this branch [version](https://github.com/NREL/EnergyPlus/blob/22cc14d218f4fa28d074125c72783c27108d337a/src/EnergyPlus/AirLoopHVACDOAS.cc#L470) of this file side by side.  This is _a little_ better to read and understand now.  And DRYed up some duplicate blocks.